HPX build failing with HWLOC errors on POWER8 with hwloc 1.8

## Expected Behavior

As of 10 hours ago, the build was fine, but one of the most recent pull requests has broken the build.

## Actual Behavior

HPX build fails with:
```
/home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/src/hpx/src/runtime/threads/topology.cpp:521:42: error: use of undeclared identifier 'HWLOC_MEMBIND_BYNODESET'
                    1, nodeset, &policy, HWLOC_MEMBIND_BYNODESET);
                                         ^
/home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/src/hpx/src/runtime/threads/topology.cpp:1236:55: error: use of undeclared identifier 'HWLOC_MEMBIND_BYNODESET'
            (hwloc_membind_policy_t)(policy), flags | HWLOC_MEMBIND_BYNODESET);
                                                      ^
/home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/src/hpx/src/runtime/threads/topology.cpp:1246:42: error: use of undeclared identifier 'HWLOC_MEMBIND_BYNODESET'
            topo, addr, len, ns, policy, HWLOC_MEMBIND_BYNODESET);
                                         ^
/home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/src/hpx/src/runtime/threads/topology.cpp:1279:47: error: use of undeclared identifier 'HWLOC_MEMBIND_BYNODESET'
                topo, addr, len, ns, &policy, HWLOC_MEMBIND_BYNODESET) == -1)
                                              ^
4 errors generated.
make[3]: *** [src/CMakeFiles/hpx.dir/runtime/threads/topology.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [src/CMakeFiles/hpx.dir/all] Error 2
make[1]: *** [CMakeFiles/core.dir/rule] Error 2
make: *** [core] Error 2
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

  1. cmake -DCMAKE_BUILD_TYPE=Release -DBOOST_ROOT=/home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/build-centaur-ppc64le-Linux-clang/boost-1.65.0 -DTCMALLOC_ROOT=/usr/local/packages/gperftools/2.5 -DHPX_WITH_MALLOC=tcmalloc -DHWLOC_ROOT= -DCMAKE_INSTALL_PREFIX=. -DHPX_WITH_THREAD_IDLE_RATES=ON -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCEL_COALESCING=OFF -DHPX_WITH_TOOLS=ON -DHPX_WITH_APEX=TRUE -DAPEX_WITH_ACTIVEHARMONY=TRUE -DACTIVEHARMONY_ROOT=/usr/local/packages/activeharmony/4.6.0-ppc64le -DAPEX_WITH_OTF2=TRUE -DOTF2_ROOT=/usr/local/packages/otf2/2.0-ppc64le -DAPEX_WITH_PAPI=TRUE -DPAPI_ROOT=/usr/local/packages/papi/5.5.1 -DHPX_WITH_APEX_NO_UPDATE=TRUE -DHPX_WITH_APEX_TAG=develop -DHWLOC_ROOT=/usr/local/packages/hwloc/1.8 -DHPX_WITH_RDTSCP=OFF -DHPX_WITH_DEPRECATION_WARNINGS=Off -DHPX_WITH_MAX_CPU_COUNT=160 /home/users/khuck/buildbot/slaves/phylanx/ppc64le-clang5-release/build/tools/buildbot/src/hpx
  1. make

## Specifications

... Please describe your environment

  - HPX Version:  current master
  - Platform (compiler, OS):  Clang 5.0 on POWER8 linux, hwloc 1.8

--- a/src/runtime/threads/topology.cpp
+++ b/src/runtime/threads/topology.cpp
@@ -517,8 +517,13 @@ namespace hpx { namespace threads
         {
             std::unique_lock<hpx::util::spinlock> lk(topo_mtx);
             int ret =
+#if HWLOC_API_VERSION >= 0x00010b06
                 hwloc_get_area_membind(topo, reinterpret_cast<void const*>(lva),
                     1, nodeset, &policy, HWLOC_MEMBIND_BYNODESET);
+#else
+                hwloc_get_area_membind_nodeset(topo,
+                    reinterpret_cast<void const*>(lva), 1, nodeset, &policy, 0);
+#endif
 
             if (-1 != ret)
             {
@@ -838,7 +843,11 @@ namespace hpx { namespace threads
     {
         hwloc_bitmap_t cpuset  = mask_to_bitmap(mask, HWLOC_OBJ_PU);
         hwloc_bitmap_t nodeset = hwloc_bitmap_alloc();
+#if HWLOC_API_VERSION >= 0x00020000
         hwloc_cpuset_to_nodeset(topo, cpuset, nodeset);
+#else
+        hwloc_cpuset_to_nodeset_strict(topo, cpuset, nodeset);
+#endif
         hwloc_bitmap_free(cpuset);
         return std::make_shared<hpx::threads::hpx_hwloc_bitmap_wrapper>(nodeset);
     }
@@ -1232,8 +1241,15 @@ namespace hpx { namespace threads
         hwloc_bitmap_ptr bitmap,
         hpx_hwloc_membind_policy policy, int flags) const
     {
-        return hwloc_alloc_membind(topo, len, bitmap->get_bmp(),
-            (hwloc_membind_policy_t)(policy), flags | HWLOC_MEMBIND_BYNODESET);
+        return
+#if HWLOC_API_VERSION >= 0x00010b06
+            hwloc_alloc_membind(topo, len, bitmap->get_bmp(),
+                (hwloc_membind_policy_t)(policy),
+                flags | HWLOC_MEMBIND_BYNODESET);
+#else
+            hwloc_alloc_membind_nodeset(topo, len, bitmap->get_bmp(),
+                (hwloc_membind_policy_t)(policy), flags);
+#endif
     }
 
     bool topology::set_area_membind_nodeset(
@@ -1242,8 +1258,14 @@ namespace hpx { namespace threads
 #if !defined(__APPLE__)
         hwloc_membind_policy_t policy = ::HWLOC_MEMBIND_BIND;
         hwloc_nodeset_t ns = reinterpret_cast<hwloc_nodeset_t>(nodeset);
-        int ret = hwloc_set_area_membind(
-            topo, addr, len, ns, policy, HWLOC_MEMBIND_BYNODESET);
+        int ret =
+#if HWLOC_API_VERSION >= 0x00010b06
+            hwloc_set_area_membind(
+                topo, addr, len, ns, policy, HWLOC_MEMBIND_BYNODESET);
+#else
+            hwloc_set_area_membind_nodeset(topo, addr, len, ns, policy, 0);
+#endif
+
         if (ret < 0)
         {
             std::string msg = std::strerror(errno);
@@ -1275,8 +1297,14 @@ namespace hpx { namespace threads
         hwloc_membind_policy_t policy;
         hwloc_nodeset_t ns = reinterpret_cast<hwloc_nodeset_t>(nodeset->get_bmp());
 
-        if (hwloc_get_area_membind(
-                topo, addr, len, ns, &policy, HWLOC_MEMBIND_BYNODESET) == -1)
+        if (
+#if HWLOC_API_VERSION >= 0x00010b06
+            hwloc_get_area_membind(
+                topo, addr, len, ns, &policy, HWLOC_MEMBIND_BYNODESET)
+#else
+            hwloc_get_area_membind_nodeset(topo, addr, len, ns, &policy, 0)
+#endif
+            == -1)
         {
             HPX_THROW_EXCEPTION(kernel_error,
                 "hpx::threads::topology::get_area_membind_nodeset",

